src/System.IO.Pipelines/src/System/IO/Pipelines/PipeOptions.cs

```diff
@@ -12,6 +12,12 @@ namespace System.IO.Pipelines
     /// </summary>
     public class PipeOptions
     {
+        private const int DefaultMinimumSegmentSize = 2048;
+
+        private const int DefaultResumeWriterThreshold = DefaultMinimumSegmentSize * Pipe.SegmentPoolSize;
```

Comment:
Why don't we calculate these on the fly in the pipe itself so it's accurate instead of being based on defaults?

Reply:
Is it worth it? These are just some safe defaults for default options, I could've put  random numbers there but decided to connect them to something, but it's not one size fits all formula and I don't think we should codify it.

Reply:
I wouldn't have said anything if you didn't make it a formula but it looks legit. Defaults are our default segment pool.